pkgs/by-name/cl/clash-verge/package.nix

```diff
@@ -1,66 +1,116 @@
-{ lib
-, stdenv
-, fetchurl
-, dpkg
-, wrapGAppsHook3
-, autoPatchelfHook
-, clash-meta
-, openssl
-, webkitgtk
-, udev
-, libayatana-appindicator
-, nix-update-script
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  nodejs,
+  pnpm,
+  cargo,
+  rustc,
+  cargo-tauri,
+  pkg-config,
+  wrapGAppsHook3,
+  glib,
+  gtk3,
+  libsoup,
+  openssl,
+  webkitgtk,
+  libayatana-appindicator,
+  clash-meta,
+  dbip-country-lite,
+  v2ray-geoip,
+  v2ray-domain-list-community,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "clash-verge";
-  version = "1.3.8";
+  version = "1.6.4";
 
-  src = fetchurl {
-    url = "https://github.com/zzzgydi/clash-verge/releases/download/v${version}/clash-verge_${version}_amd64.deb";
-    hash = "sha256-kOju4yaa+EKzFWDrk0iSJVoWkQMBjQG3hKLfAsqlsy8=";
+  src = fetchFromGitHub {
+    owner = "clash-verge-rev";
+    repo = "clash-verge-rev";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-1oPDrNsL92VUXe7HRIR7zlf45K0xQFIdncC11MRvsP8=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/src-tauri";
+
+  postPatch = ''
+    substituteInPlace $cargoDepsCopy/libappindicator-sys-*/src/lib.rs \
+      --replace-fail "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
+
+    # correct path for resources
+    substituteInPlace $cargoDepsCopy/tauri-utils-*/src/platform.rs \
+       --replace-fail "\"/usr" "\"$out"
+
+    # empty files as placeholders
+    mkdir ./{sidecar,resources}
+    touch sidecar/clash-meta-{,alpha-}x86_64-unknown-linux-gnu
+    touch resources/geo{ip,site}.dat
+    touch resources/Country.mmdb
```

Comment:
Instead of putting stubs here, I thought it's better to remove sidecar and resources from the config.

```
sed -i -e '/externalBin/d' -e '/resources/d' tauri.conf.json
```

And I thought it's better to not use sidecar at all. It's designed for binary distribution and in nixpkgs it would be better if the user can choose the core easily.

```
    substituteInPlace src/core/core.rs \
      --replace-fail 'new_sidecar(clash_core)?' 'new(clash_core)'
```

so that we don't need to put the clash core in `bin`. It will be found from the path.

Reply:
It looks like a little hack, I don't really understand, it was out of my knowledge.

Reply:
I thought the core should be handled by the module instead of hardcoding here.